packages/ui/src/hooks/use-social.ts

```diff
@@ -0,0 +1,184 @@
+import { useEffect, useCallback, useContext } from 'react';
+import { useParams } from 'react-router-dom';
+
+import { invokeSocialSignIn, signInWithSocial } from '@/apis/social';
+import { generateRandomString, parseQueryParameters } from '@/utils';
+
+import PageContext from './page-context';
+import useApi from './use-api';
+
+/**
+ * Social Connector State Utility Methods
+ * @param state
+ * @param state.uunid - unique id
+ * @param state.platform - platform
+ * @param state.callbackUriScheme - callback uri scheme
+ */
+
+type State = {
+  uunid: string;
+  platform: 'web' | 'ios' | 'android';
+  callbackUriScheme?: string;
+};
+
+const storageKeyPrefix = 'social_auth_state';
+
+export const generateState = () => {
+  const uunid = generateRandomString();
+  const platform = logtoNativeSdk?.platform ?? 'web';
+  const callbackUriScheme = logtoNativeSdk?.callbackUriScheme;
+
+  const state: State = { uunid, platform, callbackUriScheme };
+
+  return Buffer.from(JSON.stringify(state)).toString('base64');
+};
+
+export const decodeState = (state: string) => {
+  try {
+    const value = JSON.parse(Buffer.from(state, 'base64').toString()) as State;
+
+    return value;
+  } catch {}
+};
+
+export const stateValidation = (state: string, connectorId: string) => {
+  const stateStorage = sessionStorage.getItem(`${storageKeyPrefix}:${connectorId}`);
+
+  return Boolean(stateStorage) && stateStorage === state;
```

Comment:
`Boolean(stateStorage)` can be omitted, as the `state` is string type, which has no overlap with a possible `null` value

Reply:
updated